sdk/python/kubeflow/training/api/training_client_test.py

```diff
@@ -962,6 +1301,30 @@ def training_client():
         yield client
 
 
+@pytest.fixture
+def training_client_for_job_logs():
```

Comment:
Why do you need separate client for get_job_logs  ?

Reply:
Because I would like to mock [get_job_pods](https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/training/api/training_client.py#L1159-L1166) and [get_job](https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/training/api/training_client.py#L1235) function which are used in [get_job_logs](https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/training/api/training_client.py#L1090) function to make the test more isolated. If I directly use the original `training_client` fixture, I might break the tests for those two functions.

Reply:
Is there a way for us to override fixture for get_job_pods tests where we can modify Mock for [`get_job_pods` and `get_job`](https://github.com/kubeflow/training-operator/blob/041bf4fcf4ad0cbe0c127eff662d94e3825698b7/sdk/python/kubeflow/training/api/training_client_test.py#L1337-L1338) ?

Reply:
>Do you mean maybe providing some parameters for overriding the original training_client to mock get_job_pods and get_job for get_job_logs tests? 

Yes, that's right, just to avoid creating another client just for this unit test. If that is not possible, we can follow your approach.

Reply:
Thanks for the clarification! I am able to reuse the same fixture with different parameters.